roomserver/storage/shared/storage.go

```diff
@@ -1154,22 +1154,24 @@ func (d *Database) GetStateEvent(ctx context.Context, roomID, evType, stateKey s
 	if roomInfo.IsStub() {
 		return nil, nil
 	}
-	eventTypeNID, err := d.EventTypesTable.SelectEventTypeNID(ctx, nil, evType)
+	eventTypeNIDMap, err := d.eventTypeNIDs(ctx, nil, []string{evType})
 	if err == sql.ErrNoRows {
 		// No rooms have an event of this type, otherwise we'd have an event type NID
 		return nil, nil
 	}
 	if err != nil {
 		return nil, err
 	}
-	stateKeyNID, err := d.EventStateKeysTable.SelectEventStateKeyNID(ctx, nil, stateKey)
+	eventTypeNID := eventTypeNIDMap[evType]
+	stateKeyNIDMap, err := d.eventStateKeyNIDs(ctx, nil, []string{stateKey})
 	if err == sql.ErrNoRows {
```

Comment:
Does `eventStateKeyNIDs()` really return `sql.ErrNoRows`? Likewise for `eventTypeNIDs()`

Reply:
`eventStateKeyNIDs` possibly yes, `eventTypeNIDs` not.